content/articles/polymorphism-in-java/index.md

```diff
@@ -0,0 +1,243 @@
+    }
+    void info(String s) {
+        System.out.println(s+": House is " +height+ " feet tall");
+    }
+}
+public class Main {
+    public static void main(String[] args) {
+        MyClass t = new MyClass(10);
+        t.info();
+        t.info("overloaded method");
+
+        //Overloaded Constructor
+        new MyClass();
+    }
+}
+```
+
+```bash
+//Output
+Building new house that is 10 feet tall
+House is 10 feet tall
+overloaded method: House is 10 feet tall
+bricks
+```
+In the example above, you can see that we created two methods, both of which share the same name `MyClass` but have different parameters. When the first class is called, it will avoid the output of the string stated since there is no specific data type stated to be returned.
+
+On the other hand, the second class, MyClass, has a parameter of an integer. When we call it, the output will not only have the printed output but will also recognize an integer included in the output. The example shows that we can have multiple methods having the same name but different parameters and that is known as method overloading.
+
+#### 2. Dynamic Polymorphism
+
+**Dynamic Polymorphism** is a type of polymorphism wherein the case when you call an overridden method, this conflict is resolved when you run the code, normally known as runtime. It can also be referred to as `run time polymorphism`.
+
+Method Overriding is an example of dynamic polymorphism.
+
+`Method overriding` is a feature where you can have a parent method and a child method. Both the parent and child methods have the same name, but the child method will tend to modify or override the parent method.
+
+Below is a snippet that explains the concept in detail.
+
+```java
+class Animal {
+    public void sound() {
+        System.out.println("Animals make sound");
+    }
+}
+class Dog extends Animal {
+    @Override
+    public void sound() {
+        System.out.println("Woof");
+    }
+}
+class Cat extends Animal {
+    @Override
+    public void sound() {
+        System.out.println("Meow");
+    }
+}
+public class Main {
+    public static void main(String[] args) {
+        Animal a = new Animal();
+        Animal b = new Dog();
+        Animal c = new Cat();
+
+        a.sound();
+        b.sound();
+        c.sound();
+    }
+}
+
+```
+```bash
+// Output
+Animals make sound
+Woof
+Meow
+```
+
+In the example above, we have two classes, the dog class, and the cat class which both extend the class animal. You can note that when we called the object dog and cat, we receive the outputs for the methods in their respective classes.
+
+This indicates that the method in the class animal was overridden by the child class. Normally, for easier understanding of the code, we use `@Override` keyword to indicate which method is overriding the parent method. This gives a good example of dynamic polymorphism.
+
+### Advantages of Polymorphism
+We are now going to look at the advantages of each of the types of polymorphism.
+
```

Comment:
You have discussed the **Dynamic Polymorphism** advantages. But what about the **Static Polymorphism**. Or am I missing something?  @Kimosop-Wakanda 
cc @mohamedgh16

Reply:
I have made the correction @ahmadmardeni1